doc/source/examples/analysis/custom_parallel_analysis.ipynb

```diff
@@ -0,0 +1,986 @@
+    "    plt.plot(col, label=label)\n",
+    "plt.legend()\n",
+    "plt.ylabel('Radius of gyration (Å)')\n",
+    "plt.xlabel('Frame');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Parallelization in a simple per-frame fashion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Frame-wise form of the function\n",
+    "\n",
+    "First, we need to modify the single-frame method to contain the `frame_index`. Then we explicitly point to the frame that need to be analyzed: `atomgroup.universe.trajectory[frame_index]`. With that, the state of all the dynamic components inside the function will change accordingly (e.g. the position of the `AtomGroup`)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def radgyr_per_frame(frame_index, atomgroup, masses, total_mass=None):\n",
+    "    \n",
+    "    atomgroup.universe.trajectory[frame_index]\n",
+    "    \n",
+    "    # coordinates change for each frame\n",
+    "    coordinates = atomgroup.positions\n",
+    "    center_of_mass = atomgroup.center_of_mass()\n",
+    "    \n",
+    "    # get squared distance from center\n",
+    "    ri_sq = (coordinates-center_of_mass)**2\n",
+    "    # sum the unweighted positions\n",
+    "    sq = np.sum(ri_sq, axis=1)\n",
+    "    sq_x = np.sum(ri_sq[:,[1,2]], axis=1) # sum over y and z\n",
+    "    sq_y = np.sum(ri_sq[:,[0,2]], axis=1) # sum over x and z\n",
+    "    sq_z = np.sum(ri_sq[:,[0,1]], axis=1) # sum over x and y\n",
+    "    \n",
+    "    # make into array\n",
+    "    sq_rs = np.array([sq, sq_x, sq_y, sq_z])\n",
+    "    \n",
+    "    # weight positions\n",
+    "    rog_sq = np.sum(masses*sq_rs, axis=1)/total_mass\n",
+    "    # square root and return\n",
+    "    return np.sqrt(rog_sq)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parallelization with multiprocessing\n",
+    "\n",
+    "The native parallel module in python is called [multiprocessing](https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing), it contains some useful tools to build a pool of working cores, map the function into different workers, and gather and order the results from all the workers.\n",
+    "\n",
+    "Below we use `Pool` from `multiprocessing` as a context manager, we can define how many cores (or workers) we want to use by `Pool(n_jobs)`. Then we can map our analysis method `radgyr_per_frame` with the parameters needed with `partial`. Note we do not include `frame_index` inside partial, because `frame_index` is the variable that is iterated to create a list of jobs.\n",
+    "\n",
+    "The downside of `multiprocessing` is that it is mostly focused on single-machine multicore parallelism (without extra manager). It is hard to operate on multimachine conditions (we use `dask` instead) "
```

Comment:
Not sure this fits here, after the description of the function but before the actual code. It may work better as an intro to the dask section?